treeherder/model/error_summary.py

```diff
@@ -33,6 +35,19 @@ def get_error_summary(job, queryset=None):
     if cached_error_summary is not None:
         return cached_error_summary
 
+    # add support for error line caching
+    line_cache_key = 'error_lines'
+    line_cache = cache.get(line_cache_key)
+    if line_cache is None:
+        line_cache = {str(job.submit_time.date()): {}}
+    else:
+        dates = list(line_cache.keys())
+        dates.sort()
+        for d in dates:
+            dTime = datetime.datetime.strptime(d, '%Y-%m-%d')
+            if dTime <= (job.submit_time - datetime.timedelta(days=LINE_CACHE_TIMEOUT_DAYS)):
```

Comment:
This applies only once per day. An `else` condition which adds a `break` statement would be beneficial.